### ISSUE TITLE
Support for Divina

### ISSUE THREAD

**Issue by HadrienGardeur** [readium/web#4]
While Readium Web is already capable of handling comics/manga through its EPUB FXL navigator, we'd like to go further by providing a dedicated navigator optimized specifically for image-based navigation using the [Divina profile](https://readium.org/webpub-manifest/profiles/divina) of the [Readium Web Publication Manifest](https://readium.org/webpub-manifest).

This navigator would be built on top of `<canvas>` and need to be able to handle the following type of publications

- comics
- manga
- bandes dessinées 
- continuously scrolled comics (aka webtoons)

This means that we'll need the ability to support:

- pagination (LTR or RTL, over single pages or speads)
- continuously-scrolled reading (in any direction, over single pages, spreads or tiles)
- and [guided navigation](https://github.com/readium/architecture/tree/sync-media/models/guided) (probably in a later iteration and separate Epic)

On the server-side, the go-toolkit should be able to parse and serve as Divina:

- EPUB FXL (specific rules for this inference are still TBD at this point)
- CBZ files
- ZIP files containing images

In addition, we'll also need to support obfuscation over individual images of these publications to enable the distribution of some copyrighted content (this will need to be covered in a separate Epic).

**Comment by llemeurfr** [comment]
Shouldn't we consider that the job is done (or well engaged) via https://github.com/readium/divina-player-js?

**Comment by HadrienGardeur** [comment]
No, because we want a navigator aligned with what we have in Readium Mobile which isn't the case of https://github.com/readium/divina-player-js.

This means support for the Preferences API among other things.

Keep in mind that this is an Epic as well, which covers work across both toolkits.

**Comment by chocolatkey** [comment]
Just to add, regarding being "build on top of `<canvas>`", I would not recommend we go for an approach where the reading engine itself is in a `<canvas>`. We can use a canvas to render individual pages, but things like transitions, scrolling etc. should be implemented using mostly CSS. We'll save a lot of time not reinventing the wheel. Most comic readers I have observed that using a purely canvas-based engine are usually lacking many features due to the complexity involved. Not to mention accessibility will become more difficult
